app/services/youtube_service.rb

```diff
@@ -2,18 +2,27 @@ module YoutubeService
   include HTTParty
 
   KEY = ENV['GOOGLE_API_KEY']
-
-  def self.get_first_video_id(query)
+  
+  # Relevant Youtube api response example: 
+  # {"items"=>
+  #   [{"id"=>{"kind"=>"youtube#video", "videoId"=>"z1gn-CRc0qQ"},
+  #     "snippet"=>
+  #      { "thumbnails"=>
+  #        {"default"=>{"url"=>"https://i.ytimg.com/vi/z1gn-CRc0qQ/default.jpg", "width"=>120, "height"=>90},
+  #         "medium"=>{"url"=>"https://i.ytimg.com/vi/z1gn-CRc0qQ/mqdefault.jpg", "width"=>320, "height"=>180},
+  #         "high"=>{"url"=>"https://i.ytimg.com/vi/z1gn-CRc0qQ/hqdefault.jpg", "width"=>480, "height"=>360}} 
+  #       }}]}
+  def self.download_track_data(query)
     url = "https://www.googleapis.com/youtube/v3/search?key=#{KEY}&type=video&part=snippet&maxResults=1&q=#{query}".gsub("\u{d8}", "")
```

Comment:
@zinosama `\u{d8}` is the Ruby source code for [the latin capital letter o with a stroke through it](http://graphemica.com/%C3%98). I was running into trouble getting YouTube results for the song [Major Lazer & DJ Snake - Lean On (feat. MØ)](https://www.youtube.com/watch?v=YqeW9_5kURI) because it wasn't an ASCII character.